nolearn/lasagne/utils.py

```diff
@@ -0,0 +1,179 @@
+import operator as op
+
+from lasagne.layers import Conv2DLayer
+from lasagne.layers import MaxPool2DLayer
+try:
+    from lasagne.layers.cuda_convnet import Conv2DCCLayer
+    from lasagne.layers.cuda_convnet import MaxPool2DCCLayer
+except ImportError:
+    Conv2DCCLayer = Conv2DLayer
+    MaxPool2DCCLayer = MaxPool2DLayer
+import numpy as np
+from tabulate import tabulate
+
+
+class _list(list):
+    pass
+
+
+class _dict(dict):
+    def __contains__(self, key):
+        return True
+
+
+class ansi:
+    BLUE = '\033[94m'
+    CYAN = '\033[36m'
+    GREEN = '\033[32m'
+    MAGENTA = '\033[35m'
+    RED = '\033[31m'
+    ENDC = '\033[0m'
+
+
+def layers_have_conv2d(layers):
+    try:
+        return any([isinstance(layer, (Conv2DLayer, Conv2DCCLayer))
+                    for layer in layers])
+    except TypeError:
+        return isinstance(layers, (Conv2DLayer, Conv2DCCLayer))
```

Comment:
Instead of relying on this `TypeError`, can we just check for `isinstance(layers, Layer)` to determine if it's a single layer or not?  Seems friendlier.

The function name could be changed as well, maybe `has_conv2d` so it makes more sense when I call it with a single layer.